### ISSUE TITLE
DependencyResolutionDepth=DIRECT for sourceDependency does not seem to work

### ISSUE THREAD

**Issue by joh-ivu** [ascopes/protobuf-maven-plugin#164]
In a company-internal project I have the following setup:

The FooBar project shall publish some .proto files
* Those protos depend on other protos coming from Maven dependencies. There are three dependencies in total:
  * com.google.api.grpc:proto-google-common-protos (a jar with both .class and .proto files)
  * my.company:stuff (a zip with .proto files)
  * my.company:things (a zip with .proto files)
* There are transitive dependencies between them: stuff depends both on things and proto-google-common-protos. All of it depends on com.google.protobuf:protobuf-java (a jar with both .class and .proto files, namely the well known types like wrappers.proto)

I want to invoke protoc with the FooBar .proto files as an automated sanity check. I'm using the version 1.2.0-rc1 of the plugin and the following configuration:

```xml
<plugin>
    <groupId>io.github.ascopes</groupId>
    <artifactId>protobuf-maven-plugin</artifactId>
    <version>1.2.0-rc1</version>
            
    <executions>
        <execution>
            <id>compile-foobar-protos</id>
            <phase>generate-test-sources</phase>
            <goals>
                <goal>generate-test</goal>
            </goals>
            <configuration>
                <protocVersion>${protobuf.version}</protocVersion>
                <ignoreProjectDependencies>true</ignoreProjectDependencies>
                <importDependencies>
                    <importDependency>
                        <groupId>com.google.api.grpc</groupId>
                        <artifactId>proto-google-common-protos</artifactId>
                        <version>${proto-google-common-protos.version}</version>
                    </importDependency>
                </importDependencies>
                <sourceDependencies>
                    <sourceDependency>
                        <groupId>my.company</groupId>
                        <artifactId>stuff</artifactId>
                        <version>${stuff.version}</version>
                        <classifier>protobuf</classifier>
                        <type>zip</type>
                        <dependencyResolutionDepth>DIRECT</dependencyResolutionDepth>
                    </sourceDependency>
                    <sourceDependency>
                        <groupId>my.company</groupId>
                        <artifactId>things</artifactId>
                        <version>${things.version}</version>
                        <classifier>protobuf</classifier>
                        <type>zip</type>
                        <dependencyResolutionDepth>DIRECT</dependencyResolutionDepth>
                    </sourceDependency>
                </sourceDependencies>
                <sourceDirectories>
                    <sourceDirectory>path/to/foobar.proto</sourceDirectory>
                </sourceDirectories>
                <outputDirectory>${project.build.directory}/generated-test-sources/java</outputDirectory>
            </configuration>
        </execution>
    </executions>
</plugin>
```

Executing this, I see protoc errors that the .proto files from stuff.zip  and things.zip cannot be found. They are found when I remove the dependencyResolutionDepth, but then I get the other error that wrappers.proto is both on the import path (because it is a transitive dependency of the commons-proto dependency) and a source to compile (because it is a transitive dependency of stuff & things)

**Comment by ascopes** [comment]
Thanks for raising!

It's strange, the versions of protobuf that I've tested against appear to not care if things are on the import path and source path... but I may be missing something here.

What version of protoc are you using, out of curiosity?

**Comment by joh-ivu** [comment]
I'm using 3.25.3 as proto-google-common-protos is still not compatible with 4.x

**Comment by ascopes** [comment]
Cool... I'll try to look into this when I get some time then. 

If you do happen to have some way of reproducing this in an integration test or a working example, that'd be greatly appreciated, just because I want to ensure it definitely caters to how you are expecting the configuration to work. 

If that isn't possible, I can try and work something out for sure. It might just take a little longer.

**Comment by ascopes** [comment]
I've fixed the issue with DIRECT scope handling in 32df8a5 via GH-171. That appears to just be a bug with the logic that was an edge case. I've added a reproduction of that. Going to look into the transitive issue you mentioned as well and see if I can reproduce it.

**Comment by ascopes** [comment]
Okay, found a reproduction for the transitive issue finally. Will close this issue as the main problem is solved and raise a separate one to track the transitive issue.

**Comment by ascopes** [comment]
v1.2.0 has been introduced to address this main issue. The transitive dependency issue has been moved to GH-172 which I will try to work on... not sure on the best way of dealing with this situation at the moment so it will be addressed or marked wontfix if not possible to fix in the future once I've done some experimentation and investigation.

Hope this unblocks you for now!
